database/deployment_test.go

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package database
+
+import (
+	"database/sql"
+	"reflect"
+	"testing"
+
+	"github.com/go-vela/types/library"
+	"github.com/go-vela/types/raw"
+	"github.com/google/go-cmp/cmp"
+	"github.com/lib/pq"
+)
+
+func TestDatabase_Deployment_Nullify(t *testing.T) {
+	// setup types
+	var d *Deployment
+
+	want := &Deployment{
+		ID:          sql.NullInt64{Int64: 0, Valid: false},
+		Number:      sql.NullInt64{Int64: 0, Valid: false},
+		RepoID:      sql.NullInt64{Int64: 0, Valid: false},
+		URL:         sql.NullString{String: "", Valid: false},
+		Commit:      sql.NullString{String: "", Valid: false},
+		Ref:         sql.NullString{String: "", Valid: false},
+		Task:        sql.NullString{String: "", Valid: false},
+		Target:      sql.NullString{String: "", Valid: false},
+		Description: sql.NullString{String: "", Valid: false},
+		Payload:     nil,
+		CreatedAt:   sql.NullInt64{Int64: 0, Valid: false},
+		CreatedBy:   sql.NullString{String: "", Valid: false},
+		Builds:      nil,
+	}
+
+	// setup tests
+	tests := []struct {
+		deployment *Deployment
+		want       *Deployment
+	}{
+		{
+			deployment: testDeployment(),
+			want:       testDeployment(),
+		},
+		{
+			deployment: d,
+			want:       nil,
+		},
+		{
+			deployment: new(Deployment),
+			want:       want,
+		},
+	}
+
+	// run tests
+	for _, test := range tests {
+		got := test.deployment.Nullify()
+
+		if !reflect.DeepEqual(got, test.want) {
+			t.Errorf("Nullify is %v, want %v", got, test.want)
+		}
+	}
+}
+
+func TestDatabase_Deployment_ToLibrary(t *testing.T) {
+	want := new(library.Deployment)
+	want.SetID(1)
+	want.SetNumber(1)
+	want.SetRepoID(1)
+	want.SetURL("https://github.com/github/octocat/deployments/1")
+	want.SetCommit("1234")
+	want.SetRef("refs/heads/main")
+	want.SetTask("deploy:vela")
+	want.SetTarget("production")
+	want.SetDescription("Deployment request from Vela")
+	want.SetPayload(raw.StringSliceMap{"foo": "test1"})
+	want.SetCreatedAt(1)
+	want.SetCreatedBy("octocat")
+	want.SetBuilds(nil)
+
+	got := testDeployment().ToLibrary(nil)
+	if diff := cmp.Diff(got, want); diff != "" {
+		t.Errorf("(ToLibrary: -want +got):\n%s", diff)
```

Comment:
The order of want and got have to match in the error message as they do in the function call